linux/setup_repo.sh

```diff
@@ -3,8 +3,40 @@
 # This script installs a Jamulus repository to Debian based systems
 
 if [[ ${EUID} -ne 0 ]]; then
-     echo "Error: This script must be run as root."
-     exit 1
+    echo "Error: This script must be run as root."
+    exit 1
+fi
+
+# Check for apt version >= 2.2.0 (if found, assuming Debian-based compatible with repo)
+# Issue: https://bugs.launchpad.net/ubuntu/+source/apt/+bug/1950095
+
+if APT_VERSION="$(apt --version)"; then
+    APT_MAJOR=$(echo "$APT_VERSION" | cut -d' ' -f 2 | cut -d'.' -f 1)
+    APT_MINOR=$(echo "$APT_VERSION" | cut -d' ' -f 2 | cut -d'.' -f 2)
+else
+    echo "This script is only compatible with Debian based distributions which have apt, but apt is not available. Please check that your OS is supported."
+    exit 1
+fi
+
+if (($(echo "${APT_MAJOR}.${APT_MINOR} < 2.2" | bc -l))); then
```

Comment:
```
peter@fs-peter:~$ echo "2.12 < 2.2" | bc -l
1
peter@fs-peter:~$ echo "2.3 < 2.2" | bc -l
0
```
That doesn't look right to me.  I think you're after something like `$(( 2 < 2 || ( 2 == 2 && 12 < 2 ) ))`:
```suggestion
if $(( ${APT_MAJOR} < 2 || ( ${APT_MAJOR} = 2 && ${APT_MINOR} < 2 )); then
```

Reply:
Using `bc` is a shortcut because it avoids double integer calculations and a bracketed AND/OR clause which I couldn't get my head around at the time I coded this. It avoids double comparison and I found it more elegant, but your example sticks to a bash paradigm without relying on `bc`. 

Happy to defer to your way and avoid `bc` if you prefer

Reply:
More fun:
```
$ { echo 2.12 mine; echo 2.2 test; } | sort -V
2.2 test
2.12 mine
$ { echo 2.1 mine; echo 2.2 test; } | sort -V
2.1 mine
2.2 test
~$ { echo 2.3 mine; echo 2.2 test; } | sort -V
2.2 test
2.3 mine
```
from which you could do
```
peter@fs-peter:~$ if [[ "$({ echo 2.12 mine; echo 2.2 test; } | sort -V | tail -1)" == "2.2 test" ]]; then echo bad; else echo good; fi
good
peter@fs-peter:~$ if [[ "$({ echo 2.1 mine; echo 2.2 test; } | sort -V | tail -1)" == "2.2 test" ]]; then echo bad; else echo good; fi
bad
peter@fs-peter:~$ if [[ "$({ echo 2.3 mine; echo 2.2 test; } | sort -V | tail -1)" == "2.2 test" ]]; then echo bad; else echo good; fi
good
```

Reply:
It fails, though.  It doesn't handle version numbering.  So it's not a solution here.

Reply:
That's indeed why I split just major and minor. But now you mention it, 2.11 < 2.2 but could mean MINOR version 11 which is > MINOR VERSION 2. So you are correct my system is not right.